google/cloud/bigtable/test_proxy/CMakeLists.txt

```diff
@@ -12,21 +12,52 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# A minimal CMakeList.txt showing how to use the Cloud Bigtable test proxy
-# in CMake-based projects.
+include(CompileProtos)
+google_cloud_cpp_find_proto_include_dir(PROTO_INCLUDE_DIR)
 
-cmake_minimum_required(VERSION 3.10...3.24)
-project(google-cloud-cpp-bigtable-test-proxy CXX)
+google_cloud_cpp_grpcpp_library(
+    google_cloud_cpp_bigtable_test_proxy_protos
+    test_proxy.proto
+    PROTO_PATH_DIRECTORIES
+    ${PROTO_INCLUDE_DIR}
+    ${PROJECT_SOURCE_DIR}
+    ${PROJECT_SOURCE_DIR}/google/cloud/bigtable/test_proxy
+    ${PROJECT_BINARY_DIR}/external/googleapis/src/googleapis_download)
+target_link_libraries(
+    google_cloud_cpp_bigtable_test_proxy_protos
+    PUBLIC google-cloud-cpp::api_client_protos
+           google-cloud-cpp::bigtable_protos
+           google-cloud-cpp::rpc_status_protos)
+set_target_properties(google_cloud_cpp_bigtable_test_proxy_protos
+                      PROPERTIES CXX_CLANG_TIDY "")
 
-find_package(google_cloud_cpp_bigtable REQUIRED)
+add_library(
+    cbt_test_proxy
+    cbt_test_proxy.cc
+    cbt_test_proxy.h)
+target_link_libraries(
+    cbt_test_proxy
+    google_cloud_cpp_bigtable_test_proxy_protos
+    google-cloud-cpp::bigtable
+    google-cloud-cpp::bigtable_protos
+    google-cloud-cpp::common
+    google-cloud-cpp::grpc_utils
+    gRPC::grpc++
+    gRPC::grpc
+    protobuf::libprotobuf)
 
-# MSVC requires some additional code to select the correct runtime library
-if (VCPKG_TARGET_TRIPLET MATCHES "-static$")
-    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>")
-else ()
-    set(CMAKE_MSVC_RUNTIME_LIBRARY "MultiThreaded$<$<CONFIG:Debug>:Debug>DLL")
+google_cloud_cpp_add_executable(target "cbt_test_proxy_main" "cbt_test_proxy_main.cc")
```

Comment:
nit: I would say
```
google_cloud_cpp_add_executable(target "bigtable_test_proxy" "cbt_test_proxy_main.cc")
```

Basically cmake targets need to be unique and what that function does is concatenate the prefix with the file name, to create a target. So I think we generally want the prefix to look like the directory structure.

So with my suggestion, the target would be: `bigtable_test_proxy_cbt_test_proxy_main`.